web/packages/teleport/src/services/mfa/makeMfaDevice.ts

```diff
@@ -30,12 +30,16 @@ export default function makeMfaDevice(json): MfaDevice {
     description = 'unknown device';
   }
 
+  const type = json.type === 'TOTP' ? 'totp' : 'webauthn';
+  const usage = residentKey ? 'passwordless' : 'mfa';
+
   return {
     id,
     name,
     description,
     registeredDate: new Date(addedAt),
     lastUsedDate: new Date(lastUsed),
-    residentKey,
```

Comment:
Note: dropping this field broke e/ builds.

Example: https://github.com/gravitational/teleport.e/actions/runs/8173863278/job/22347267175

```
yarn run v1.22.21
$ NODE_OPTIONS='--max-old-space-size=4096' tsc
e/web/teleport/src/Recovery/RecoveryFlow/RecoveryFlow.test.tsx(7[5](https://github.com/gravitational/teleport.e/actions/runs/8173863278/job/22347267175#step:8:6),11): error TS2353: Object literal may only specify known properties, and 'residentKey' does not exist in type 'MfaDevice'.
e/web/teleport/src/Recovery/RecoveryFlow/RecoveryFlow.test.tsx(83,11): error TS2353: Object literal may only specify known properties, and 'residentKey' does not exist in type 'MfaDevice'.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 2.
```

@bl-nero